### PR TITLE
Add more communication kernels

### PR DESCRIPTION
**Pull request by Binyang2014**
Add more communication kernels
- Packet related kernels
- Fused kernel for recv-reduce-send

### CODE REVIEW

File: ark/include/kernels/comm.h [hunk 1]
```diff
@@ -100,7 +257,7 @@ DEVICE void read(int ChanId, size_t remote_offset, size_t local_offset,
     DataType *local_data = reinterpret_cast<DataType *>(local);
     DataType *remote_data = reinterpret_cast<DataType *>(remote);
     DefaultBroadcast1<InDims, InShape, DataType, OutDims, OutShape, DataType,
-                      type::Identity, true, false, UnitOutDims, NumWarps,
+                      type::Identity, false, false, UnitOutDims, NumWarps,
```
Comment (chhwang) [inline ark/include/kernels/comm.h:260]: Does this improve throughput?

File: examples/tutorial/multi_gpu_plan.py
Comment (chhwang) [inline examples/tutorial/multi_gpu_plan.py:1]: Seems plan files are missing

File: ark/include/kernels/comm.h (hunk 1 above)
Comment (Binyang2014) [reply ark/include/kernels/comm.h:260]: Yes, use non atomic read/write will improve the performance.

File: examples/tutorial/multi_gpu_plan.py
Comment (Binyang2014) [reply examples/tutorial/multi_gpu_plan.py:1]: Okay, let me check-in the example plan files

Comment (Binyang2014) [reply examples/tutorial/multi_gpu_plan.py:1]: added

**Pull request merged**